flexibleengine/resource_flexibleengine_dli_queues_v1.go

```diff
@@ -0,0 +1,244 @@
+package flexibleengine
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+
+	"github.com/huaweicloud/golangsdk/openstack/common/tags"
+	"github.com/huaweicloud/golangsdk/openstack/dli/v1/queues"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var regexp4Name = regexp.MustCompile(`^[a-z0-9_]+$`)
+
+func ResourceDliQueueV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDliQueueCreate,
+		Read:   resourceDliQueueV1Read,
+		Delete: resourceDliQueueV1Delete,
+
+		Schema: map[string]*schema.Schema{
+			"queue_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					if !regexp4Name.MatchString(v) {
+						errs = append(errs, fmt.Errorf("%q can contain only digits, lower letters, and underscores (_) ", key))
+					}
+					return
+				},
+			},
+
+			"queue_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "sql",
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"cu_count": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"charging_mode": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Default:  1,
+			},
+
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "0",
+				StateFunc: func(val interface{}) string {
+					if val.(string) == "" {
+						return "0"
+					} else {
+						return val.(string)
+					}
+				},
+			},
+
+			"platform": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "x86_64",
+			},
+
+			"resource_mode": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"labels": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"feature": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "basic",
+			},
+
+			"tags": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"cidr_in_vpc": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```

Comment:
suggest to adding to validate functions  of `queue_type`, `charging_mode`, `platform`, `resource_mode`, `feature`.

Reply:
ok, added